### DISCUSSION TITLE
Validate related fields in a cue schema?

### DISCUSSION THREAD

**Discussion by cbrgm** [cue-lang/cue#1143] [Q&A] [answered]
hey, I’m trying my hand at a cue schema to validate a configuration.
```
{
 foo: "something" 
 bar: "somethingelse"
}
```
is it possible to define dependencies between fields? 

AC: 
* The validation should fail (`exit code 1 / error`) if either only `foo` or only `bar` is set.
* It should return `exit code 0 / ok` if either **neither** or **both** have been set. 
* In other words: XOR with `foo` and `bar` as inputs 

Does anyone have an example?

**Comment by verdverm** [comment, marked as answer]
You can use a disjunction like this `#XOR: {} | {a: _, b: _}`, however care has to be taken with fields and values generally.

https://cuelang.org/play/?id=weHb_EEW-N3#cue@export@yaml  (note how I had to explicitly close the empty struct value)

**Reply by cbrgm** [reply]
Awesome!  That's a good example! Thanks @verdverm

**Reply by myitcv** [reply]
Related are proposals #822 and #943, specifically `numexist`, which would handle the edge case that @verdverm flags above.
